src/Compilers/Core/Portable/Collections/StaticCast.cs

```diff
@@ -10,7 +10,7 @@ namespace Microsoft.CodeAnalysis
 {
     internal static class StaticCast<T>
     {
-        internal static ImmutableArray<T> From<TDerived>(ImmutableArray<TDerived> from) where TDerived : class?, T
+        internal static ImmutableArray<T> From<TDerived>(ImmutableArray<TDerived> from) where TDerived : class, T
```

Comment:
📝 The original appears to be equally correct here. Can you provide more context for this specific change?

Reply:
This is how it's annotated in the framework.

Reply:
Let's keep our form here (with a suppression) and file a bug on the framework. It appears to be a simple omission when NRT was enabled.

Reply:
As a simple example of what should be possible:

```csharp
ImmutableArray<object> values = ImmutableArray.Create(new object());
ImmutableArray<object?> nullableValues = StaticCast<object?>.From(values);
```

Reply:
https://github.com/dotnet/runtime/issues/39799